python/semantic_kernel/agents/chat_completion_agent.py

```diff
@@ -0,0 +1,207 @@
+
+        messages = await chat_completion_service.get_chat_message_contents(
+            chat_history=chat,
+            settings=settings,
+            kernel=self.kernel,
+            arguments=KernelArguments(),
+        )
+
+        logger.info(
+            f"[{type(self).__name__}] Invoked {type(chat_completion_service).__name__} "
+            f"with message count: {message_count}."
+        )
+
+        # Capture mutated messages related function calling / tools
+        for message_index in range(message_count, len(chat)):
+            message = chat[message_index]
+            message.name = self.name
+            history.add_message(message)
+
+        for message in messages:
+            message.name = self.name
+            yield message
+
+    @override
+    async def invoke_stream(  # type: ignore
+        self, history: ChatHistory
+    ) -> AsyncGenerator[StreamingChatMessageContent, None]:
+        """Invoke the chat history handler in streaming mode.
+
+        Args:
+            kernel: The kernel instance.
+            history: The chat history.
+
+        Returns:
+            An async generator of StreamingChatMessageContent.
+        """
+        # Get the chat completion service
+        chat_completion_service = self.kernel.get_service(service_id=self.service_id, type=ChatCompletionClientBase)
+
+        if not chat_completion_service:
+            raise KernelServiceNotFoundError(f"Chat completion service not found with service_id: {self.service_id}")
+
+        assert isinstance(chat_completion_service, ChatCompletionClientBase)  # nosec
+
+        settings = (
+            self.execution_settings
+            or self.kernel.get_prompt_execution_settings_from_service_id(self.service_id)
+            or chat_completion_service.instantiate_prompt_execution_settings(
+                service_id=self.service_id, extension_data={"ai_model_id": chat_completion_service.ai_model_id}
+            )
+        )
+
+        chat = self._setup_agent_chat_history(history)
+
+        message_count = len(chat)
+
+        logger.debug(f"[{type(self).__name__}] Invoking {type(chat_completion_service).__name__}.")
+
+        messages: AsyncGenerator[list[StreamingChatMessageContent], Any] = (
+            chat_completion_service.get_streaming_chat_message_contents(
+                chat_history=chat,
+                settings=settings,
+                kernel=self.kernel,
+                arguments=KernelArguments(),
+            )
+        )
+
+        logger.info(
+            f"[{type(self).__name__}] Invoked {type(chat_completion_service).__name__} "
+            f"with message count: {message_count}."
+        )
+
+        # Capture mutated messages related function calling / tools
+        for message_index in range(message_count, len(chat)):
+            message = chat[message_index]
+            message.name = self.name
+            history.add_message(message)
+
+        async for message_list in messages:
```

Comment:
will the final completion message be added to the history here in the streaming case?